src/pages/Shop/index.js

```diff
@@ -0,0 +1,382 @@
+
+  changeValueView = item => {
+    return this.setState({ valueDropView: item });
+  };
+
+  changeValueDate = item => {
+    return this.setState({ valueDropDate: item, valueDropPrice: "" });
+  };
+}
+
+const types = ["all", "chair", "sofa", "storage", "beds", "lamps"];
+
+const ListTypes = ({ changeParams, type, params }) => {
+  return (
+    <ul className={styles.listType}>
+      <h3 className={styles.typeTitle}>Categories</h3>
+      {types.map((item, index) => (
+        <li
+          onClick={() =>
+            changeParams({
+              ...params,
+              type: item === "all" ? undefined : item,
+              _start: 0,
+              _q: undefined
+            })
+          }
+          className={
+            type === item || (!type && item === "all")
+              ? `${styles.typeItemActive} ${styles.typeItem}`
+              : `${styles.typeItem}`
+          }
+          key={index}
+        >
+          {item}
+        </li>
+      ))}
+    </ul>
+  );
+};
+
+const CartList = ({ list, params, addToCart, cartList }) => {
+  if (isEmpty(list) || !list[stringifySearchParams(params)]) {
+    return <Loading />;
+  }
+  const searchList = list[stringifySearchParams(params)];
+  return (
+    <div className={styles.cartList}>
+      {searchList.map(({ title, price, slider, id }, index) => (
+        <div key={index} className={styles.item}>
+          <Link to={`/product/${id}`} className={styles.content}>
+            <img className={styles.img} src={slider[0].large} alt="product" />
+          </Link>
+          <div className={styles.imgData}>
+            <p className={styles.price}>{price}</p>
+            <h4 className={styles.title}>{title}</h4>
+            <div
+              onClick={() => addToCart(id, slider[1], title, price)}
+              className={
+                cartList[id]
+                  ? `${styles.basketActive} ${styles.basket}`
+                  : `${styles.basket}`
+              }
+            >
+              <Basket />
+            </div>
+          </div>
+        </div>
+      ))}
+    </div>
+  );
+};
+
+const listDropDownView = [MIN_VIEW_SHOP, AVERAGE_VIEW_SHOP, MAX_VIEW_SHOP];
+
+const listDropDownSort = ["desc", "asc"];
+
+const DropDown = ({
+  changeParams,
+  dropMenuIsOpen,
+  arr,
+  changeParamsValue,
+  changeParamsName,
+  changeValueDropDown,
+  params
+}) => {
+  return (
+    <div
+      className={
+        !dropMenuIsOpen
+          ? `${styles.dropMenuListClose}`
+          : `${styles.dropMenuList}`
+      }
+    >
+      {arr.map((item, index) => (
+        <p
+          className={styles.dropMenuItem}
+          onClick={() =>
+            changeParams(
+              (changeParamsName === "_sort" && {
+                ...params,
+                _sort: `${changeParamsValue}${item}`
+              }) ||
+                (changeParamsName === "_limit" && {
+                  ...params,
+                  _limit: `${changeParamsValue}${item}`
+                })
```

Comment:
use  object assign it will looks better than ternary operators

```
Object.assign(params,
 changeParamsName === "_sort" && _sort: `${changeParamsValue}${item}`,
 changeParamsName === "_limit" && _limit: `${changeParamsValue}${item}`
)
```